crates/devices/virtio-vsock/src/packet.rs

```diff
@@ -460,19 +460,30 @@ impl<'a, B: BitmapSlice> VsockPacket<'a, B> {
             return Err(Error::InvalidHeaderLen(pkt.len()));
         }
 
-        let data_desc = desc_chain.next().ok_or(Error::DescriptorChainTooShort)?;
+        // Starting from Linux 6.2 the virtio-vsock driver can use a single descriptor for both
+        // header and data.
+        let data_slice = if chain_head.len() >= PKT_HEADER_SIZE as u32 + pkt.len() {
+            mem.get_slice(
+                chain_head
+                    .addr()
+                    .checked_add(PKT_HEADER_SIZE.try_into().unwrap())
```

Comment:
nit: Since we are doing a `try_into()` followed by an `unwrap` I think we can just use `PKT_HEADER_SIZE as u64`. This is safe since the PKT_HEADER_SIZE is a usize. Unwraps tend to make you think that something potentially dangerous us going on 😆

Reply:
ack, I'll change it!